validations/src/validations.rs

```diff
@@ -744,7 +744,12 @@ pub fn validate_commit_transaction(
 
     let dr_output = &dr_state.data_request;
 
-    // A value transfer output cannot have zero value
+    // Commitment's output is only for change propose, so it only has to be one output and the
+    // address has to be the same than the address which creates the commitment
+    let proof_pkh = co_tx.body.proof.proof.pkh();
+    if co_tx.body.outputs.len() > 1 {
+        return Err(TransactionError::SeveralCommitOutputs.into());
+    }
     for (idx, output) in co_tx.body.outputs.iter().enumerate() {
```

Comment:
This for can be converted into an if-let, because there can only be 0 or 1 outputs. And idx is always 0.

```suggestion
    if let Some(output) = &co_tx.body.outputs {
```